Source/SwiftLintFramework/Rules/Idiomatic/NoMagicNumbersRule.swift

```diff
@@ -0,0 +1,85 @@
+import SwiftSyntax
+
+public struct NoMagicNumbersRule: SwiftSyntaxRule, OptInRule, ConfigurationProviderRule {
+    public init() {}
+
+    public var configuration = SeverityConfiguration(.warning)
+
+    public static let description = RuleDescription(
+        identifier: "no_magic_numbers",
+        name: "No Magic Numbers",
+        description: "Magic numbers should be replaced by named constants.",
+        kind: .idiomatic,
+        nonTriggeringExamples: [
+            Example("var foo = 123"),
+            Example("static let bar: Double = 0.123"),
+            Example("array[0] + array[1] "),
+            Example("let foo = 1_000.000_01"),
+            Example("// array[1337]"),
+            Example("baz(\"9999\")"),
+            Example("""
+        func foo() {
+            let x: Int = 2
+            let y = 3
+            let vector = [x, y, -1]
+        }
+        """),
+            Example("""
+        class A {
+            var foo: Double = 132
+            static let bar: Double = 0.98
+        }
+        """),
+            Example("""
+        @available(iOS 13, *)
+        func version() {
+            if #available(iOS 13, OSX 10.10, *) {
+                return
+            }
+        }
+        """)
+        ],
+        triggeringExamples: [
+            Example("foo(↓321)"),
+            Example("bar(↓1_000.005_01)"),
+            Example("array[↓42]"),
+            Example("let box = array[↓12 + ↓14]"),
+            Example("let a = b + 2.0"),
```

Comment:
```suggestion
            Example("let a = b + ↓2.0"),
```
And could you also add an example like `let a = b + 1.0` to the non-triggering section?